### PR TITLE
Space Vietnam Grass Coat

### PR DESCRIPTION
**Pull request by gurfan**
Botany syndicate item. Costs 6 TC normally, 4 TC for botanists.
Looks identical to the grass coat (mino) found in biogenerators.
While wearing the coat, space vines won't entangle you, inject you, bite you, sting you, etc.
Vines **will** still block your vision, so thermals/mesons/xray will still be helpful.

:cl:
 * rscadd: The syndicate received a shipment of authentic Space Vietnamese grass coats and have added them to agent uplinks. Agents wearing the coat won't be affected by aggressive plant life.

### CODE REVIEW

File: code/modules/clothing/suits/miscellaneous.dm [hunk 1]
```diff
@@ -818,6 +818,11 @@ obj/item/clothing/suit/cassock
 	body_parts_covered = ARMS|FULL_TORSO|IGNORE_INV
 	species_fit = list(VOX_SHAPED, INSECT_SHAPED)
 
+/obj/item/clothing/suit/mino/vinesafe
+
+/obj/item/clothing/suit/mino/vinseafe/vine_protected()
```
Comment (Kurfursten) [inline code/modules/clothing/suits/miscellaneous.dm]: if you typo the type path it won't work at all

also you don't need to have the blank definition above, it gets implied by the overridden proc

Comment (gurfan) [reply code/modules/clothing/suits/miscellaneous.dm]: both fixed

File: code/modules/hydroponics/spreading/spreading_response.dm [hunk 2]
```diff
@@ -154,23 +162,33 @@
...
 		if(istype(H.shoes, /obj/item/clothing/shoes/magboots) && (H.shoes.clothing_flags & NOSLIP))
 			can_grab = 0
+		if(is_protected(victim))
+			can_grab = 0
 	if(can_grab)
 		src.visible_message("<span class='danger'>Tendrils lash out from \the [src] and drag \the [victim] in!</span>")
 		victim.forceMove(src.loc)
 		lock_atom(victim, /datum/locking_category/plantsegment)
 
+/obj/effect/plantsegment/proc/is_protected(var/mob/living/carbon/human/H)
```
Comment (DamianX) [inline code/modules/hydroponics/spreading/spreading_response.dm]: why is this a proc on `/obj/effect/plantsegment`?

Comment (gurfan) [reply code/modules/hydroponics/spreading/spreading_response.dm]: There's no good reason why. I've moved the proc to /mob/living

File: code/modules/hydroponics/spreading/spreading_response.dm [hunk 3]
```diff
@@ -154,23 +162,34 @@
...
 		src.visible_message("<span class='danger'>Tendrils lash out from \the [src] and drag \the [victim] in!</span>")
 		victim.forceMove(src.loc)
 		lock_atom(victim, /datum/locking_category/plantsegment)
 
+/mob/living/proc/vine_protected()
+	return FALSE
+
+/mob/living/carbon/human/vine_protected()
+	var/obj/item/clothing/suit/S = wear_suit
+	if(!S || !istype(S))
```
Comment (DamianX) [inline code/modules/hydroponics/spreading/spreading_response.dm]:
```suggestion
	if(!istype(S))
```

**Comment by xeroxemnas** [comment]
HE'S IN THE VINES, BLAST HIM

**Comment by Eneocho** [comment]
Can't wait for the eventual round where everyone gets a space vietnam coat and continue playing as normal despite the cyanide mindbreaker kudzu covering all the station.

File: code/datums/uplink_item.dm [hunk 4]
```diff
@@ -880,6 +880,14 @@ var/list/uplink_items = list()
...
 	jobs_with_discount = list("Botanist")
 
+/datum/uplink_item/jobspecific/service/vinesuit
+	name = "Space Vietnam Grass Coat"
+	desc = "This inconspicuous grass coat was woven from kudzu fibers for guerilla missions in Space Vietnam. While wearing the coat, space vines won't entangle, bite, or otherwise harm you."
+	item = 	/obj/item/clothing/suit/mino/vinesafe
+	cost = 6
+	discounted_cost = 4
+	jobs_with_discount = list("Botanist")
+
```
Comment (jknpj) [inline code/datums/uplink_item.dm:890]:
```suggestion
mech_flags = MECH_SCAN_ILLEGAL
```

Comment (jknpj) [reply code/datums/uplink_item.dm:890]: This isn't the right place to put it. I just wanted to remind you about it.

**Pull request merged**